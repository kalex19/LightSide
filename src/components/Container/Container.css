.card-container {
    align-items: center;
    display: flex;
    flex-wrap: wrap;
    margin: auto;
}

.back-btn {
    color: white;
    display: block;
    font-size: 2em;
    margin: 10px 42%;
    text-decoration: none;
}

.back-btn:visited {
    text-decoration: none;
}

.back-btn:hover {
    color: yellow;
}

.showMoreBtn {
    color: white;
    background-color: turquoise;
    border-radius: 15px;
    font-family: 'Raleway';
    font-size: 16px;
    font-weight: 900;
    margin: 2% 42%;
    opacity: 0.8;
    padding: 10px;
    width: 150px;
}