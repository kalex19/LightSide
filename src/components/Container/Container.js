import React from 'react';
import Card from '../Card/Card.js';
import './Container.css';

const Container = ({
    data
}) => {
    const findData = data.map(item => {
        {console.log(data)}

        return <Card
<<<<<<< HEAD
            name={item.name}
            birthyear={item.birth_year}
            gender={item.gender}
            height={item.height}
            eyecolor={item.eye_color}
            model={item.model}
            terrain={item.terrain}
            diameter={item.diameter}
            population={item.population}
            class={item.vehicle_class}
            passengers={item.passengers}
=======
        name = {
            item.name
        }
        birthyear = {
            item.birth_year
        }
        gender = {
            item.gender
        }
        height = {
            item.height
        }
        eyecolor = {
            item.eye_color
        }
        model = {
            item.model
        }
        class = {
            item.vehicle_class
        }
        passengers = {
            item.passengers
        }
        favorites = {
            data.favorites
        }
>>>>>>> 69e111519ed6b53133c5c53841204500a45ca389
        />
    })
    return ( <
        main className = "card-container" > {
            findData
        } <
        /main>
    )
}

export default Container;