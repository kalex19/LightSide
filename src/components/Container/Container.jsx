import React from 'react';
import Card from '../Card/Card';
import './Container.css';
import PropTypes from 'prop-types';

const Container = ({ data, favoriteCard }) => {
	const findData = data.map(item => {
		console.log(item);
		return <Card info={item} key={item.created} favoriteCard={favoriteCard} />;
	});

	return <main className="card-container">{findData}</main>;
};

Container.propTypes = {
	data: PropTypes.array,
	favoriteCard: PropTypes.func
};

export default Container;
