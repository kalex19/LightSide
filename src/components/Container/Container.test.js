import React from 'react';
import { shallow, mount } from 'enzyme';
import Container from './Container.jsx';

describe('Container', () => {
    it('should match snapshot', () => {
        const mockData = [
            {
                class: "wheeled",
                id: "2014-12-10T15:36:25.724000Z",
                model: "Digger Crawler",
                name: "Sand Crawler",
                passengers: "30"
            }
        ];
        const wrapper = shallow(
            <Container 
                data={mockData} 
                favoriteCard={jest.fn()} 
                handleClick={jest.fn()}
                num='4'
            />)
        
        expect(wrapper).toMatchSnapshot();
    })

    it('should show more cards when showMore is clicked', () => {
        const mockCallBack = jest.fn();
        const mockData =[{
                class: "wheeled",
                id: "4",
                model: "Digger Crawler",
                name: "Sand Crawler",
                passengers: "30"
            }];
        const wrapper = shallow(
            <Container 
                data={mockData} 
                favoriteCard={jest.fn()} 
                handleClick={jest.fn()}
                num={4}
            />)

        mockCallBack();
        wrapper.find('.showMoreBtn').simulate('click');
        expect(mockCallBack).toHaveBeenCalled()

    })

    
})