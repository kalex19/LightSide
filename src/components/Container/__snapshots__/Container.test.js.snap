// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match snapshot 1`] = `
<main
  className="card-container"
>
  <Card
    favoriteCard={[MockFunction]}
    info={
      Object {
        "class": "wheeled",
        "id": "2014-12-10T15:36:25.724000Z",
        "model": "Digger Crawler",
        "name": "Sand Crawler",
        "passengers": "30",
      }
    }
  />
  <Link
    className="back-btn"
    to="/"
  >
    <i
      class="fab fa-jedi-order"
    >
       Back Home 
    </i>
  </Link>
  <button
    className="showMoreBtn"
    onClick={[Function]}
  >
    Show More
  </button>
</main>
`;
